Add Advanced Section (#164)
R2Northstar/NorthstarWiki: docs/using-northstar/advanced.md

# Advanced Options

{% hint style="warning" %}

> This is a section detailing the more advanced parts of using Northstar, with some of them having a chance to potentially mess with your game. Setting your level too high, for example, will lock you out of the game until you reset your entire player stats, progress, and loadouts.

Reviewer: ```suggestion
This is a section detailing the more advanced parts of using Northstar, with some of them having a chance to potentially mess with your game. Setting your level too high, for example, will lock you out of Northstar's multiplayer until you reset your entire player stats, progress, and loadouts.
```
It only blocks from multiplayer, not singleplayer

In order to create the `.bat` to launch this profile, you'll need to first create a `.txt` file inside of your Titanfall2 directory, and name it whatever you would like. Inside of this `.txt` file, you'll want to put in `NorthstarLauncher.exe -profile=PROFILE FOLDER NAME HERE`, replacing all of `PROFILE FOLDER NAME HERE` with the name that you gave the folder you installed your mods to. After this, rename the file to `yourFileName.bat`. Double clicking or right clicking on the `.bat` file then hitting `open` will launch Northstar with the assigned profile.

You can set up profiles in an even more advanced way by setting up a way to use Steam to launch multiple different profiles from their newer launch menu that appears when you can launch a game in more ways than one. This is also covered [here](installing-northstar/basic-setup.mdbasic-setup#adding-alternate-launch-option-for-steam), and can be set up for profiles by simply adding the `-profile=PROFILE FOLDER NAME HERE` to the arguments of the new option for NorthstarLauncher, as seen below

> ![SteamEdit using Northstar Profiles](images/steamedit-profiles.png)

Reviewer 2: Image path also needs to be updated :P

```suggestion
![SteamEdit using Northstar Profiles](../images/steamedit-profiles.png)
```

> ![SteamEdit using Northstar Profiles](images/steamedit-profiles.png)

Reviewer 2: Image path also needs to be updated :P

```suggestion
![SteamEdit using Northstar Profiles](../images/steamedit-profiles.png)
```

### Vanilla Profile

> A simple "vanilla" profile can be created using a `.bat` file, without needing to fully make an additional mods folder. You can do this simply by creating a `.txt` file inside of your Titanfall2 directory (you can name it whatever you want), and putting in `NorthstarLauncher.exe -norestrictservercommands -profile=R2Vanilla`. After doing this, you'll want to rename it to `yourFileName.bat`. This tells the NorthstarLauncher to not restrict server commands, which is what normally disables the Vanilla multiplayer servers on Northstar to lower confusion, and tells it to use the `R2Vanilla` profile, which, as it doesn't normally exist, will launch Northstar with no core mods enabled, allowing you to easily play on Vanilla using Northstar's security fixes.

Reviewer 2: `-norestrictservercommands` is actually to prevent server running certain commands on client. Disabled on Northstar cause here server cannot be trusted but needed for vanilla to work.

```suggestion
A simple "vanilla" profile can be created using a `.bat` file, without needing to fully make an additional mods folder. You can do this simply by creating a `.txt` file inside of your Titanfall2 directory (you can name it whatever you want), and putting in `NorthstarLauncher.exe -norestrictservercommands -profile=R2Vanilla`.
After doing this, you'll want to rename it to `yourFileName.bat`.
This tells the NorthstarLauncher to not restrict server commands, a feature which is needed to allow the server to tell the client which server to join (but is disabled on Northstar for security reasons), and tells it to use the `R2Vanilla` profile, which, as it doesn't normally exist, will launch Northstar with no core mods enabled, allowing you to easily play on Vanilla using most of Northstar's security fixes.
```

Double clicking the `.bat` or right clicking on it and hitting `open` will launch the vanilla profile.

## Setting levels using console commands <a href="#set-level" id="set-level"></a>

{% hint style="warning" %}

> This section here can cause you to mess up your persistence (save data) on Northstar if you set your levels too high! Make sure to read carefully!

Reviewer: ```suggestion
This section here can cause you to mess up your persistence (multiplayer user data) on Northstar if you set your levels too high! Make sure to read carefully!
```
minor correction, semantics but we do still need to distinguish multiplayer problems from singleplayer ones

{% endhint %}

> With everything unlocked, there is no need to set your level to a higher level, but some users may stil want to do so. In order to this, you'll need to open the console and type in the relevant commands. Both of these require `sv_cheats 1` to be enabled on the server you're playing on, easily done by using `sv_cheats 1` in the console in a private match. 

Reviewer: ```suggestion
With everything unlocked, there is no need to set your level to a higher level, but some users may still want to do so. In order to this, you'll need to open the in-game console with the `~` button on your keyboard and type in the relevant commands. Both of these require `sv_cheats 1` to be enabled on the server you're playing on, easily done by using `sv_cheats 1` in the console in a private match. 
```
we should clarify they need to open the in-game console and not try to use the console window that pops up while Northstar is running (also typo cleanup)

If you experience strange issues after using these, you probably set something too high, and should follow the [resetting levels wiki section](installing-northstar/troubleshooting.md#i-used-a-command-to-set-my-playergun-xp-level-and-i-set-it-too-high-so-now-my-game-crashes-when-trying-to-join-multiplayer).

## Playing Vanilla via NorthstarLauncher <a href="#vanilla-on-northstar" id="vanilla-on-northstar"></a>

It is recommended that you [set up a vanilla profile](advanced.md#profiles) instead of disabling all of your mods and using `-norestrictservercommands` as a launch option, however if you wish to only play vanilla using this method and would rather not set up a profile, you can do the following.

The reason behind doing this is that Northstar has several security fixes that Vanilla does not have, however these are not *necessary*. The odds you get hacked playing vanilla are close to zero, and there are no reports of people being genuinely hacked by playing Vanilla.\

> This method assumes you're launching Northstar via Titanfall 2 on EA/Steam/Origin using launch options

Reviewer: ```suggestion
This method assumes you're launching Northstar via Titanfall 2 on EA or Steam using launch options.
```
We should stop mentioning Origin in the wiki since its deprecated by EA.
